### DISCUSSION TITLE
search domain

### DISCUSSION THREAD

**Discussion by alexkayzer** [coredns/coredns#6075] [Q&A]
Hi! Please tell me, faced with such a problem:
I have an external domain example.com ,
i have "someone" service install outside the k8s,
**core DNS has the setting**
```
    .:53 {
        errors
        metadata .
        health {
            lameduck 5s
        }
        ready
        log . "{type} {class} {name} ns:{/kubernetes/client-namespace} pod:{/kubernetes/client-pod-name} status:{rcode}"
        prometheus :9153
        kubernetes cluster.local in-addr.arpa ip6.arpa {
            pods verified
            fallthrough in-addr.arpa ip6.arpa
        }
        forward . /etc/resolv.conf
        cache 30
        loop
        reload 5s
        loadbalance
    }
```
**resolv.conf:**
```
search default.svc.cluster.local svc.cluster.local cluster.local example.com
nameserver 10.160.4.10
options ndots:5
```
if I want to access the service outside for example
```
ping someone
PING someone (10.2.140.88): 56 data bytes
```
is good
**log:**
```
[INFO] AAAA IN someone.svc.cluster.local. ns:default pod:busybox status:NXDOMAIN
[INFO] A IN someone.svc.cluster.local. ns:default pod:busybox status:NXDOMAIN
[INFO] A IN someone.example.com. ns:default pod:busybox status:NOERROR
[INFO] AAAA IN someone.example.com. ns:default pod:busybox status:NOERROR 
```
But if i install "someone" service in cluster i have error: 
```
/ # ping someone
ping: bad address 'someone'
```
**log:**
```
[INFO] AAAA IN someone.svc.cluster.local. ns:default pod:busybox status:NXDOMAIN
[INFO] A IN someone.svc.cluster.local. ns:default pod:busybox status:NXDOMAIN
[INFO] A IN someone.example.com. ns:default pod:busybox status:NXDOMAIN
[INFO] AAAA IN someone.example.com. ns:default pod:busybox status:NXDOMAIN 
```
in "sameone" namespace resolve working good, outside - not working
Why as soon as I have a service in the cluster  ```[INFO] AAAA IN someone.example.com. ns:default pod:busybox status:NXDOMAIN``` ?????

**Comment by chrisohaver** [comment]
>[INFO] A IN someone.example.com. ns:default pod:busybox status:NXDOMAIN

Per the coredns config, this NXDOMAIN response must be coming from upstream (10.160.4.10).  CoreDNS just passes the response along.

> But if i install "someone" service in cluster ...

In what namespace did you install the "someone" service?

> in "sameone" namespace resolve working good, outside - not working

I'm struggling to understand what you mean by this.  Can you rephrase?

**Reply by alexkayzer** [reply]
"someone" service deployed in "someone" namespace. 
I have "someone" service installed out of k8s( on VMs) and i have "sometwo" service installed in k8s on nemaspace "sometwo".
"sometwo" get from "someone" a few data. it's worked. after install "someone" in to k8s 
"sometwo" can't get access to "someone"....
i have A record "someone.example.com" in DNS outside k8s. k8s know about it.
someone.example.com - it a balancer address. Balance to vms and k8s-ingress nodes.
outside cluster all worked good.

**Reply by chrisohaver** [reply]
> "sometwo" get from "someone" a few data. it's worked.

seems directly contradictory of ...

> "sometwo" can't get access to "someone"

**Reply by alexkayzer** [reply]
If i add "namespaces" in config map coredns ( config exluded namespace "someone" - right?)
```
       kubernetes cluster.local in-addr.arpa ip6.arpa {
            pods verified
            namespaces default, kube-system
            fallthrough in-addr.arpa ip6.arpa
        }
```
it's working! request going to away from cluster! 
But....
I can't access to this namespace directly or cross-namespace (via someone.someone.svc don't working to)

**Comment by chrisohaver** [comment]
> [INFO] AAAA IN someone.example.come. ns:default pod:busybox status:NOERROR 

`example.come.` ... I presume this is a typo resulting from cleansing the logs of private domain info.  Perhaps some information is lost in that cleansing?

**Reply by alexkayzer** [reply]
it's mistake on copy-paste) example.com - right
